torch/fx/passes/utils/matcher_utils.py

```diff
@@ -301,7 +376,16 @@ def match(self, graph: Graph) -> List[InternalMatch]:
         def backtracking(anchor_index, match):
             if anchor_index == len(match_candidates_list):
                 match.placeholder_nodes = [match.nodes_map[pn] for pn in self.pattern_placeholder_nodes]
-                match.returning_nodes = [match.nodes_map[pn] for pn in self.pattern_returning_nodes]
+                match.returning_nodes = []
+                for pn in self.pattern_returning_nodes:
+                    ret_nodes = match.nodes_map[pn]
+                    if isinstance(ret_nodes, list):
+                        # This is assuming that the module only returns one
+                        # thing (or the last node in the flattened module nodes
+                        # is the return value)
+                        match.returning_nodes.append(ret_nodes[-1])
```

Comment:
Does this work for things like nn.MaxPool2D? can you check? I believe the aten op is max_pool2d_with_indices but we discard indices. Or perhaps would it be better to create subgraph out of `match.nodes_map[pn]` and inspect its ret nodes? It is a little bit harder in this case perhaps because the value returned by MaxPool2D `call_module` can return either pooled tensor or tuple of (pooled_tensor, indices)